boltdesignsystem/bolt
# [6.x]: DS-823 add examples for linear and radial gradients backgrounds

## Jira

https://pegadigitalit.atlassian.net/browse/DS-823

## Summary

After the discussion, we decide to move ahead and deprecate a few of the background props.

## Details

- `fill_color` prop is deprecated
- `color` enum of the `fill` prop was renamed to `solid`
- `linear-gradient` and `radial-gradient` enums were deprecated (`solid` and `gradient` can be used)

**A solid overlay on the background component can be achieved by adding theming class `t-bolt-*` to the parent of a background component. The gradient option doesn't have support for this yet.**

## How to test

Pull the branch. Check if props are properly deprecated or renamed. Confirm that achieving solid overlay can be done by passing the theme class to the parent of the background component.

## Release notes

### Fill Prop Changes
- The `fill` prop option `color` was renamed to `solid`.
- The `fill` props options `linear-gradient` and `radial-gradient` were removed. These are now handled with a new `.u-bolt-bg-gradient` class, refer to the [Background Gradients](https://boltdesignsystem.com/pattern-lab/?p=viewall-visual-styles-background-gradients) docs page for more information.
- The `fill_color` and `fillColor` props are now removed from the schema. Solid overlay on the background component can be achieved by adding a [theming class](https://boltdesignsystem.com/pattern-lab/?p=visual-styles-theming-system) t-bolt-* to the parent of a background component.

### Shape Prop Changes
- The `shapeGroup` prop are now removed from the schema. Use `shape_group` instead.
- The `shapeAlignment` prop are now removed from the schema. Use `shape_alignment` instead.

### Overlay Prop Changes
- The `overlay` prop is now removed from the schema. Use the `opacity` prop instead.
- Change the `overlay: enabled` prop to `opacity: medium`
- Change the `overlay: disabled` prop to `opacity: none`

### Focal Point Prop Changes
- The `focalPoint` prop is now removed from the schema. Use the `focal_point` prop instead.

### Content Items Props Changes
- The `contentItems` and `content_items` props are now removed from the schema. Use the `items` prop instead (each item must be renderable, e.g. a string or render array. Arrays intended to be rendered with `pattern_template()` are no longer supported).
